src/test/java/org/sonarlint/intellij/config/global/SonarLintGlobalSettingsTest.java

```diff
@@ -47,10 +56,89 @@ public void testRoundTrip() {
 
   @Test
   public void testGetInstance() {
-    SonarLintGlobalSettings instance = new SonarLintGlobalSettings();
-    super.register(app, SonarLintGlobalSettings.class, instance);
-    assertThat(SonarLintGlobalSettings.getInstance()).isEqualTo(instance);
+    SonarLintGlobalSettings settings = new SonarLintGlobalSettings();
+    super.register(app, SonarLintGlobalSettings.class, settings);
+    assertThat(SonarLintGlobalSettings.getInstance()).isEqualTo(settings);
+
+  }
+
+  @Test
+  public void testLoadStateOldFormat() throws Exception {
+    SonarLintGlobalSettings state = new SonarLintGlobalSettings();
+    HashSet<String> includedRules = new HashSet<>();
+    includedRules.add(RULE);
+    HashSet<String> excludedRules = new HashSet<>();
+    excludedRules.add(RULE1);
+    Field includedRulesField = state.getClass().getDeclaredField("includedRules");
```

Comment:
Don't use reflection. We usually relax encapsulation by allowing classes in the same package to access attributes for testing purposes.